com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl

```diff
@@ -343,13 +346,38 @@ half AdditionalLightRealtimeShadow(int lightIndex, float3 positionWS, half3 ligh
     return SampleShadowmap(TEXTURE2D_ARGS(_AdditionalLightsShadowmapTexture, sampler_AdditionalLightsShadowmapTexture), shadowCoord, shadowSamplingData, shadowParams, true);
 }
 
-half GetShadowFade(float3 positionWS)
+half GetMainLightShadowFade(float3 positionWS)
+{
+    // This code computes distance between capsule and point.
+    // Finally distance is used with shadow fade scale and bias to compute shadow fade blend value.
+
+    float3 shadowFadeCenter = _CascadeShadowSplitSphereLast.xyz;
+    float3 worldSpaceCameraOppositeDir = -UNITY_MATRIX_V[2].xyz; // Move it to other property for faster access?
```

Comment:
Use GetWorldSpaceViewDir instead https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl#L94
 
 It's more efficient in case of perspective projection.

Reply:
Looking at code, it seems like it is going to be a bit heavier.

// Computes the world space view direction (pointing towards the viewer).
float3 GetWorldSpaceViewDir(float3 positionWS)
{
    if (IsPerspectiveProjection())
    {
        // Perspective
        return GetCurrentViewPosition() - positionWS;
    }
    else
    {
        // Orthographic
        return -GetViewForwardDir();
    }
}